### PR TITLE
FIX-#5927: Enable `test_map_metadata` test on the HDK engine

### LINKED ISSUE #5927: BUG: Enable test_map_metadata test on the HDK engine and add to ci
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
Run test_map_metadata on the HDK engine.
```

### Issue Description

There are 25 failures, when the test is running on the HDK engine.

### Expected Behavior

All these failures should be fixed and the test added to CI.

### Error Logs

<details>

```python-traceback

Replace this line with the error backtrace (if applicable).

```

</details>

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by AndreyPavlenko**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5927? 
- [x] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
@AndreyPavlenko CI is failed.

**Comment by AndreyPavlenko** [comment]
> @AndreyPavlenko CI is failed.

Seems the failures are caused by some infrastructure issues.

**Comment by anmyachev** [comment]
> > @AndreyPavlenko CI is failed.
> 
> Seems the failures are caused by some infrastructure issues.

I restarted CI

File: modin/pandas/test/dataframe/test_map_metadata.py [hunk 1]
```diff
@@ -714,10 +718,10 @@ def test_drop():
...
     df_equals(modin_simple.drop([0, 3], axis="index"), simple.loc[[1, 2], :])
 
-    pytest.raises(ValueError, modin_simple.drop, 5)
-    pytest.raises(ValueError, modin_simple.drop, "C", 1)
-    pytest.raises(ValueError, modin_simple.drop, [1, 5])
-    pytest.raises(ValueError, modin_simple.drop, ["A", "C"], 1)
+    pytest.raises(KeyError, modin_simple.drop, 5)
+    pytest.raises(KeyError, modin_simple.drop, "C", 1)
+    pytest.raises(KeyError, modin_simple.drop, [1, 5])
+    pytest.raises(KeyError, modin_simple.drop, ["A", "C"], 1)
```
Comment (anmyachev) [inline modin/pandas/test/dataframe/test_map_metadata.py:724]: Why?

Comment (anmyachev) [reply modin/pandas/test/dataframe/test_map_metadata.py:724]: Got it, according to docs it is right: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.drop.html?highlight=drop#pandas.DataFrame.drop

File: modin/pandas/dataframe.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def __init__(
                 # the DataFrame and sets columns to the columns provided.
                 elif columns is not None and data.name not in columns:
                     self._query_compiler = from_pandas(
-                        self.__constructor__(columns=columns)
+                        pandas.DataFrame(columns=columns)
```
Comment (anmyachev) [inline modin/pandas/dataframe.py:147]: Good catch!

File: modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py [hunk 3]
```diff
@@ -981,8 +996,9 @@ def _union_all(
         if (
             len(other_modin_frames) == 0
             or len(self.columns) == 0
-            or any((len(f.columns) != len(self.columns) for f in other_modin_frames))
-            or any((set(f._dtypes) != set(self._dtypes) for f in other_modin_frames))
+            or all(f._has_arrow_table() for f in ([self] + other_modin_frames))
```
Comment (anmyachev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py:999]: Why `_has_arrow_table` is used here?

Comment (AndreyPavlenko) [reply modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py:999]: If all frames have an arrow table, we perform the arrow-based concatenation. There is a related issue - https://github.com/modin-project/modin/pull/5932.

Comment (anmyachev) [reply modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py:999]: ok, thanks for the clarification!

**Pull request merged**